url82787aa7a4_vrthra_v_language_tgz-pJ8-v_VJ8/src/v/java/Helper.java

```diff
@@ -0,0 +1,212 @@
+        Quote q = tq.qvalue();
+        Stack<Class> st = new Stack<Class>();
+        st.push(Object.class);
+        for(Term t: q.tokens()) {
+            Class c = getJavaType(t);
+            if (c.isPrimitive() && !st.peek().isAssignableFrom(c))
+                st.push(c);
+            else {
+                while(st.size() > 0) {
+                    Class p = st.peek();
+                    if (!p.isAssignableFrom(c))
+                        st.pop();
+                    else{
+                        if (!p.equals(c))
+                            st.push(c);
+                        break;
+                    }
+                }
+            }
+        }
+        return st.pop();
+    }
+
+    public static Class getArrayType(Term tq) {
+        return Array.newInstance(getArrayElementType(tq), 0).getClass();
+    }
+
+    public static Object getArrayObj(Term tq) {
+        Class c = getArrayElementType(tq);
+        Object arr = Array.newInstance(c, tq.size());
+        int i = 0;
+
+        for(Term t: tq.qvalue().tokens()) {
+            Array.set(arr, i, getJavaObj(t));
+            i++;
+        }
+        return arr;
+    }
+
+    public static Class getJavaType(Term t) {
+        switch (t.type) {
+            case TBool:
+                return Boolean.TYPE;
+            case TChar:
+                return Character.TYPE;
+            case TInt:
+                return Integer.TYPE;
+            case TDouble:
+                return Double.TYPE;
+            case TString:
+                return String.class;
+            case TObject:
+                return t.ovalue().getClass();
+            case TQuote:
+                return getArrayType(t);
+            default:
+                throw new VException("err:java:type",t,"Unable to convert type :" + t.value());
+        }
+    }
+
+    public static Object getJavaObj(Term t) {
+        switch (t.type) {
+            case TBool:
+                return new Boolean(t.bvalue());
+            case TChar:
+                return new Character(t.cvalue());
+            case TInt:
+                return new Integer(t.ivalue());
+            case TDouble:
+                return new Double(t.dvalue());
+            case TString:
+                return new String(t.svalue());
+            case TObject:
+                return t.ovalue();
+            case TQuote:
+                return getArrayObj(t);
+            default:
+                throw new VException("err:java:obj",t, "Unable to convert value :" + t.value() + " " + t.type);
+        }
+    }
+
+    // Convert a term to the most closest java object.
+    // for arrays, convert to the LCM of the available classes
+    public static Object[] getParams(Quote q) {
+        LinkedList<Object> arr = new LinkedList<Object>();
+        for (Term t: q.tokens())
+            arr.add(getJavaObj(t));
+        return arr.toArray(new Object[0]);
+    }
+
+    public static Term getResult(Object o) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.